overlord/servicestate/servicemgr.go

```diff
@@ -132,9 +133,15 @@ func (m *ServiceManager) ensureSnapServicesUpdated() (err error) {
 	}
 
 	rewrittenServices := make(map[*snap.Info][]*snap.AppInfo)
+	serviceKillingMightHaveOccurred := false
 	observeChange := func(app *snap.AppInfo, unitType, name string, old, new string) {
 		if unitType == "service" {
 			rewrittenServices[app.Snap] = append(rewrittenServices[app.Snap], app)
+			if !serviceKillingMightHaveOccurred {
+				if strings.Contains(old, "\nRequires=usr-lib-snapd.mount\n") {
```

Comment:
this is the conditional check, I add to adjust quite a few tests to still be sensible